src/ray/gcs/gcs_client/service_based_gcs_client.cc

```diff
@@ -52,6 +52,32 @@ void ServiceBasedGcsClient::Disconnect() {
   RAY_LOG(INFO) << "ServiceBasedGcsClient Disconnected.";
 }
 
+uint64_t ServiceBasedGcsClient::Reconnect(uint64_t reconnect_count) {
+  absl::MutexLock lock(&mutex_);
```

Comment:
How is this function supposed to be called? And should this be async instead?

Reply:
Gcs client will be called by multiple threads, but it only has one rpc client, so sync is better.